### DISCUSSION TITLE
Blocking bots from viewing the GitHub repository

### DISCUSSION THREAD

**Discussion by faasterweb** [community/community#72591] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

Hello. I want to block all bots from viewing a repository in my GitHub. And I want only Google bots to be able to view my repository. How can I do it? Thank you in advance for your help.

**Comment by oussamaef06** [comment]
GitHub provides a built-in feature to control access to your repositories by limiting access to known web crawlers and search engines. You can configure this feature to allow only Googlebot (Google's web crawler) to access your repository while blocking other bots. Here's how you can do it:

Go to Repository Settings:

Navigate to your GitHub repository where you want to configure access.
Access Control Settings:

In the repository, click on the "Settings" tab.
Scroll Down to the GitHub Pages Section:

GitHub Pages provides the option to control access to your repository based on web crawlers.
Configure Pages Site Settings:

In the "GitHub Pages" section, scroll down to the "GitHub Pages site settings" and look for the "Web crawler access" option.
Choose "Allow" for Googlebot:

In the "Web crawler access" dropdown, select "Allow" for "Googlebot," which is the user-agent for Google's web crawler. This allows Googlebot to access your repository.
Choose "Disallow" for Other Bots:

In the same dropdown, select "Disallow" for "All other web crawlers." This will block all other bots.
Save Changes:

After configuring these settings, scroll down to the bottom of the page and click "Save repository details" to save your changes.
Please be aware that this setting will instruct GitHub Pages to configure a robots.txt file for your repository, which is a standard way to communicate with web crawlers. Googlebot respects this file, but other bots will be discouraged from crawling your repository.

Keep in mind that while this setting can block most web crawlers, it's not a foolproof method to restrict access. Some web crawlers may not respect the robots.txt file, and determined malicious bots might still attempt to access your repository. Also, be sure to keep your Googlebot settings up to date, as they may change over time.

**Reply by faasterweb** [reply]
Hi @oussamaef06 . As far as I understand, I can set up my site on the xxx.github.io address with this solution. Is there a way to put the site in the github.com repository and block the entry of unauthorized crawlers?

**Reply by oussamaef06** [reply]
Certainly!

**Reply by faasterweb** [reply]
@oussamaef06 can you tell me?

**Reply by faasterweb** [reply]
@oussamaef06 ????

**Reply by oussamaef06** [reply]
yes you are right :)
